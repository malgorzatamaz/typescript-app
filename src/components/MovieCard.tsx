import { Card, CardBody, CardTitle, CardText } from "reactstrap";
import { Link } from "react-router-dom";
import React from "react";
import styled from "styled-components";

import { Movie } from "types/movieTypes";
import margins from "assets/margins";

export interface MovieCardProps {
  movie: Movie;
}

const StyledCard = styled(Card)`
  margin-bottom: ${margins.m};
  margin-right: ${margins.m};
  width: calc(20% - ${margins.m});
`;

const MovieCard: React.FC<MovieCardProps> = ({ movie }) => {
  return (
    <StyledCard>
      <img src={movie.image} alt={movie.title} />
      <CardBody>
        <CardTitle tag="h6">{movie.title}</CardTitle>
        <CardText>{movie.year}</CardText>
        <Link to={`/movies/${movie.id}`}>Read more</Link>
      </CardBody>
    </StyledCard>
  );
};

export default MovieCard;
