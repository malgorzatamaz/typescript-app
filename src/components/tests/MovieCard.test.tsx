import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";

import MovieCard from "components/MovieCard";

test("renders card with movie data", () => {
  const movie = { title: "Title", image: "img", id: "123", year: 2000 };

  const element = render(
    <Router>
      <MovieCard movie={movie} />
    </Router>
  );
  expect(element).toMatchSnapshot();
});
