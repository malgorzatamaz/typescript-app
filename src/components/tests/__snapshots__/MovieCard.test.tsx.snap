// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card with movie data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM bxAvtM card"
      >
        <img
          alt="Title"
          src="img"
        />
        <div
          class="card-body"
        >
          <h6
            class="card-title"
          >
            Title
          </h6>
          <p
            class="card-text"
          >
            2000
          </p>
          <a
            href="/movies/123"
          >
            Read more
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM bxAvtM card"
    >
      <img
        alt="Title"
        src="img"
      />
      <div
        class="card-body"
      >
        <h6
          class="card-title"
        >
          Title
        </h6>
        <p
          class="card-text"
        >
          2000
        </p>
        <a
          href="/movies/123"
        >
          Read more
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
